website/script/visualizer.js

```diff
@@ -288,11 +288,14 @@ function showGame(game, $container, maxWidth, maxHeight, showmovement, isminimal
                 for(var b = 0; b < game.width; b++) {
                     var site = game.frames[frame][Math.floor(loc / game.width)][loc % game.width];
                     if(site.strength == 255) mapGraphics.lineStyle(1, '0xfffff0');
-                    mapGraphics.beginFill(game.players[site.owner].color);
-                    var pw = rw * Math.sqrt(site.strength / 255) / 2, ph = rh * Math.sqrt(site.strength / 255) / 2;
-                    var move = t > 0 ? game.moves[frame][Math.floor(loc / game.width)][loc % game.width] : 0;
+                    if(site.strength != 0) mapGraphics.beginFill(game.players[site.owner].color);
+                    var pw = rw * Math.sqrt(site.strength > 0 ? site.strength / 255 : 0.1) / 2
+                    var ph = rh * Math.sqrt(site.strength > 0 ? site.strength / 255 : 0.1) / 2;
+                    var direction = frame < game.moves.length ? game.moves[frame][Math.floor(loc / game.width)][loc % game.width] : 0;
```

Comment:
hehe, so after changing this, testing with a non-square map and trying to figure out why it wouldn't work. I finally realised that no indeed they should both be game.width, (i.e. the row should be increase every game.width columns). So the current line is actually correct.

But having said that I really don't understand why the loc variable is there at all. It is only used to generate the x and y indexes into the map. The thing is those are already directly available as the for loop variables `b` and `a`.